### PR TITLE
Added new mirror-oci-artifacts job macro

### PR DESCRIPTION
**Pull request by plnordquist-pnnl**
Added a new macro to support using the latest skopeo and mirroring all oci artifacts like helm charts, etc.  Needs testing.

### CODE REVIEW

**Comment by plnordquist-pnnl** [comment]
Tested now.  Retagging helm charts will probably break them completely so it might need some docs and maybe even a check to find out if the job is attempting to do that.

File: gitlab-lib.yaml [hunk 1]
```diff
@@ -57,6 +57,68 @@
...
+    # SOURCE_REGISTRY_USERNAME - Optional username to use for the source registry
+    # SOURCE_REGISTRY_PASSWORD - Optional password to use for the source registry
+    # DEST_REGISTRY - Optional custom destination registry, defaults to CI_REGISTRY
+    # DEST_REGISTRY_PATH - Optional custom destination registry path, defaults to CI_PROJECT_PATH
+    # DEST_REGISTRY_USERNAME - Optional username to use for destination registry, defaults to CI_REGISTRY_USER
+    # DEST_REGSITRY_PASSWORD - Optional password to use for destination registry, defaults to CI_REGISTRY_PASSWORD
+    # OVERWRITE - Set this to anything to allow overwriting artifacts that already exist
+    # COPY_LOCAL - Set this to anything to copy artifacts to local disk before pushing to the destination,
+    #   useful when pulling and pushing to the same registry with separate credentials
+    image: quay.io/skopeo/stable:latest
```
Comment (kfox1111) [inline gitlab-lib.yaml]: Does latest actually have a shell now? That was a problem in the past.

We may want to consider pinning it too. If they change behavior of the container, things probably shouldn't break.

Comment (plnordquist-pnnl) [reply gitlab-lib.yaml]: I've pinned the image now.  The v1.12.0 image has /bin/sh in it and is built from a Fedora container image variant based on the /etc/os-release output.

**Review by kfox1111 — APPROVED** [review]
LGTM

**Comment by GitBytes** [comment]
> LGTM

+1

**Review by GitBytes — APPROVED** [review]

**Pull request merged**